presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -168,6 +168,41 @@ public static Slice stPoint(@SqlType(DOUBLE) double x, @SqlType(DOUBLE) double y
         return serialize(geometry);
     }
 
+    @SqlNullable
+    @Description("Returns a Geometry type MultiPoint object formed from an array of points")
+    @ScalarFunction("ST_MultiPoint")
+    @SqlType(GEOMETRY_TYPE_NAME)
+    public static Slice stMultiPoint(@SqlType("array(" + GEOMETRY_TYPE_NAME + ")") Block input)
+    {
+        MultiPoint multipoint = new MultiPoint();
+        for (int i = 0; i < input.getPositionCount(); i++) {
+            Slice slice = GEOMETRY.getSlice(input, i);
+
+            // Ignore null points
+            if (slice.getInput().available() == 0) {
+                continue;
+            }
+
+            OGCGeometry geometry = deserialize(slice);
+            if (!(geometry instanceof OGCPoint)) {
+                throw new PrestoException(INVALID_FUNCTION_ARGUMENT, format("ST_MultiPoint takes only an array of valid points, %s was passed", geometry.geometryType()));
+            }
+            OGCPoint point = (OGCPoint) geometry;
+
+            // Empty points are ignored
+            if (point.isEmpty()) {
+                continue;
+            }
+
+            multipoint.add(point.X(), point.Y());
+        }
+        if (multipoint.getPointCount() < 1) {
+            return null;
+        }
+        OGCGeometry geometry = createFromEsriGeometry(multipoint, null);
```

Comment:
nit: `geometry` variable is used only once; let's inline

`return serialize(createFromEsriGeometry(multipoint, null));`